lib/working_hours/computation.rb

```diff
@@ -82,7 +82,7 @@ def advance_to_working_time time, config: nil
         time_in_day = time.seconds_since_midnight
         config[:working_hours][time.wday].each do |from, to|
           return time if time_in_day >= from and time_in_day < to
-          return time + (from - time_in_day) if from >= time_in_day
+          return time.beginning_of_day + from if from >= time_in_day
```

Comment:
This is because `seconds_since_midnight` ignores nanoseconds, leading to a final Time leaking some nanoseconds. By using this new syntax we reset the small counters and output a clean time corresponding with the opening hour.